### DISCUSSION TITLE
Build problems on windows with VS and intel fortran

### DISCUSSION THREAD

**Discussion by aymanhab** [scivision/mumps-superbuild#80] [Q&A]
Hi all, 
Having installed Intel fortran compiler, trying to build on windows/Msys with MSVC I'm getting build failure due to missing MPI. Any idea what I'm doing wrong here or if this set of tools is supported? Note that both cl.exe and ifort.exe are located correctly and the cmakeCache file shows that mpiexec.exe was located correctly as well.

-- Building for: Visual Studio 17 2022
-- The C compiler identification is MSVC 19.37.32825.0
-- The Fortran compiler identification is Intel 2021.1.0.20231117
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.37.32822/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Determine Intel Fortran Compiler Implicit Link Path
-- Determine Intel Fortran Compiler Implicit Link Path - done
-- Check for working Fortran compiler: C:/Program Files (x86)/Intel/oneAPI/compiler/2024.0/bin/ifort.exe - skipped
-- MUMPS 5.6.2.2 upstream 5.6.2 install prefix: D:/msys64/home/ayman/mumps/mumps/build/local
-- CMake 3.23.3  Toolchain
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- Checking for module 'mpi-c'
--   Package 'mpi-c', required by 'virtual:world', not found
-- Could NOT find MPI_C (missing: MPI_C_LIB_NAMES MPI_C_WORKS)
-- Checking for module 'mpi-fort'
--   Package 'mpi-fort', required by 'virtual:world', not found
-- Could NOT find MPI_Fortran (missing: MPI_Fortran_LIB_NAMES MPI_Fortran_WORKS)
CMake Error at D:/dev/tools/CMake/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find MPI (missing: MPI_C_FOUND MPI_Fortran_FOUND C Fortran)
Call Stack (most recent call first):
  D:/dev/tools/CMake/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  D:/dev/tools/CMake/share/cmake-3.23/Modules/FindMPI.cmake:1830 (find_package_handle_standard_args)
  CMakeLists.txt:45 (find_package)

**Comment by scivision** [comment]
I would look near the end of file "build/CMakeFiles/CMakeConfigureLog.yaml" (assuming you did `cmake -B build`) to see what specifically failed in the MPI compiler test.

**Comment by scivision** [comment]
I would try using the Ninja generator, this may be why MPI isn't detected properly

```sh
cmake -B build -G Ninja
```

**Comment by aymanhab** [comment]
Thanks @scivision for your help, I tried Ninja and it failed for another reason (can't find Fortran compiler altogether despite running the batch file provided by intel to set the environment) while using the default VS finds Fortran compiler without an issue. I'd also note that  it looks like the Fortran compiler documented/supported (ifort) will be deprecated later this year so switching to LLVM fortran compiler maybe needed sooner rather than later.  Thanks again

**Reply by scivision** [reply]
I added tips on hinting the location of the Fortran compiler: https://github.com/scivision/mumps/blob/main/Readme_oneapi.md

**Comment by scivision** [comment]
@aymanhab I just fixed the Visual Studio generator, so you don't have to use Ninja anymore with the latest MUMPS.
